cylc/flow/parsec/config.py

```diff
@@ -145,8 +145,11 @@ def get(self, keys: Optional[Iterable[str]] = None, sparse: bool = False):
                     cfg = cfg[key]
                 except (KeyError, TypeError):
                     if (
+                        # __MANY__ setting not present:
                         parents in self.manyparents or
-                        key in self.spec.get(*parents)
+                        # setting not present in __MANY__ section:
+                        key in self.get(parents, sparse=False)
+                        # (expanding cfg is ok because we raise at this point)
```

Comment:
Partially reverts e7bddf8abf91d4124287514c4b64828ba1f56c42 (from #4887); at the time I was concerned by the fact that we are expanding the config here, but it shouldn't matter because the train terminates here as we're raising an exception